textdb/textdb-exp/src/main/java/edu/uci/ics/textdb/exp/nltksenti/NltkSentiOperator.java

```diff
@@ -0,0 +1,183 @@
+        }
+        this.inputOperator = operator;
+    }
+    
+    /*
+     * adds a new field to the schema, with name resultAttributeName and type String
+     */
+    private Schema transformSchema(Schema inputSchema){
+        if (inputSchema.containsField(predicate.getResultAttributeName())) {
+            throw new RuntimeException(String.format(
+                    "result attribute name %s is already in the original schema %s", 
+                    predicate.getResultAttributeName(),
+                    inputSchema.getAttributeNames()));
+        }
+        return Utils.addAttributeToSchema(inputSchema, 
+                new Attribute(predicate.getResultAttributeName(), AttributeType.STRING));
+    }
+    
+	@Override
+	public void open() throws TextDBException {
+	    if (cursor != CLOSED) {
+            return;
+        }
+        if (inputOperator == null) {
+            throw new DataFlowException(ErrorMessages.INPUT_OPERATOR_NOT_SPECIFIED);
+        }
+        inputOperator.open();
+        Schema inputSchema = inputOperator.getOutputSchema();
+        
+        // check if input schema is present
+        if (! inputSchema.containsField(predicate.getInputAttributeName())) {
+            throw new RuntimeException(String.format(
+                    "input attribute %s is not in the input schema %s",
+                    predicate.getInputAttributeName(),
+                    inputSchema.getAttributeNames()));
+        }
+        
+        // check if attribute type is valid
+        AttributeType inputAttributeType = 
+                inputSchema.getAttribute(predicate.getInputAttributeName()).getAttributeType();
+        boolean isValidType = inputAttributeType.equals(AttributeType.STRING) || 
+                inputAttributeType.equals(AttributeType.TEXT);
+        if (! isValidType) {
+            throw new RuntimeException(String.format(
+                    "input attribute %s must have type String or Text, its actual type is %s",
+                    predicate.getInputAttributeName(),
+                    inputAttributeType));
+        }
+        
+        // generate output schema by transforming the input schema
+        outputSchema = transformSchema(inputOperator.getOutputSchema());
+        
+        cursor = OPENED;
+	}
+	
+	public static void main(String[] args2){
+	    String className;
+	    String inputText = "I am glad you arrived!! :)";
+        
+        try{
+            String path = Utils.getResourcePath("Python", TextdbProject.TEXTDB_EXP);
+            path += PYTHONSCRIPT;
+            String trainner = Utils.getResourcePath("Senti.pickle", TextdbProject.TEXTDB_EXP);
+            List<String> args = new ArrayList<String>(Arrays.asList(PYTHON, path, trainner));
+            args.addAll(new ArrayList<String>(Arrays.asList(inputText)));
+            System.out.println(args);
+
+            ProcessBuilder pb = new ProcessBuilder(args);
```

Comment:
To increase the performance, let's send a batch of records to NLTK via a file.  Make the number of records in the batch as a parameter in the constructor.